pulsar-broker/src/main/java/com/yahoo/pulsar/broker/LocalBrokerData.java

```diff
@@ -0,0 +1,345 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.pulsar.broker;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import com.yahoo.pulsar.common.policies.data.loadbalancer.NamespaceBundleStats;
+import com.yahoo.pulsar.common.policies.data.loadbalancer.ResourceUsage;
+import com.yahoo.pulsar.common.policies.data.loadbalancer.ServiceLookupData;
+import com.yahoo.pulsar.common.policies.data.loadbalancer.SystemResourceUsage;
+
+/**
+ * Contains all the data that is maintained locally on each broker.
+ */
+public class LocalBrokerData extends JSONWritable implements ServiceLookupData {
+
+    // URLs to satisfy contract of ServiceLookupData (used by NamespaceService).
+    private final String webServiceUrl;
+    private final String webServiceUrlTls;
+    private final String pulsarServiceUrl;
+    private final String pulsarServiceUrlTls;
+
+    // Most recently available system resource usage.
+    private ResourceUsage cpu;
+    private ResourceUsage memory;
+    private ResourceUsage directMemory;
+
+    private ResourceUsage bandwidthIn;
+    private ResourceUsage bandwidthOut;
+
+    // Message data from the most recent namespace bundle stats.
+    private double msgThroughputIn;
+    private double msgThroughputOut;
+    private double msgRateIn;
+    private double msgRateOut;
+
+    // Timestamp of last update.
+    private long lastUpdate;
+
+    // The stats given in the most recent invocation of update.
+    private Map<String, NamespaceBundleStats> lastStats;
+
+    private int numTopics;
+    private int numBundles;
+    private int numConsumers;
+    private int numProducers;
+
+    // All bundles belonging to this broker.
+    private Set<String> bundles;
+
+    // The bundles gained since the last invocation of update.
+    private Set<String> lastBundleGains;
```

Comment:
In is not used in the load manager, but it is used in the monitor to alert users to recent bundle gains.